spec/patterns/nagios_spec.rb

```diff
@@ -0,0 +1,198 @@
+  end
+
+  it "generates the nagios_state field" do
+    expect(grok_match(subject, value)).to include("nagios_state" => "UP")
+  end
+
+  it "generates the nagios_statetype field" do
+    expect(grok_match(subject, value)).to include("nagios_statetype" => "HARD")
+  end
+
+end
+
+describe "NAGIOSLOGLINE - CURRENT SERVICE STATE" do
+
+  let(:value)   { "[1427925600] CURRENT SERVICE STATE: nagioshost;nagiosservice;OK;HARD;1;nagiosmessage" }
+  let(:grok)    { grok_match(subject, value) }
+
+  it "a pattern pass the grok expression" do
+    expect(grok).to pass
+  end
+
+  it "matches a simple message" do
+    expect(subject).to match(value)
+  end
+
+  it "generates the nagios_type field" do
+    expect(grok_match(subject, value)).to include("nagios_type" => "CURRENT SERVICE STATE")
+  end
+
+  it "generates the nagios_epoch field" do
+    expect(grok_match(subject, value)).to include("nagios_epoch" => "1427925600")
+  end
+
+  it "generates the nagios_message field" do
+    expect(grok_match(subject, value)).to include("nagios_message" => "nagiosmessage")
+  end
+
+  it "generates the nagios_hostname field" do
+    expect(grok_match(subject, value)).to include("nagios_hostname" => "nagioshost")
+  end
+
+  it "generates the nagios_service field" do
+    expect(grok_match(subject, value)).to include("nagios_service" => "nagiosservice")
+  end
+
+  it "generates the nagios_state field" do
+    expect(grok_match(subject, value)).to include("nagios_state" => "OK")
+  end
+
+  it "generates the nagios_statetype field" do
+    expect(grok_match(subject, value)).to include("nagios_statetype" => "HARD")
+  end
+
+end
+
+describe "NAGIOSLOGLINE - TIMEPERIOD TRANSITION" do
+
+  let(:value)   { "[1427925600] TIMEPERIOD TRANSITION: 24X7;1;1" }
+  let(:grok)    { grok_match(subject, value) }
+
+  it "a pattern pass the grok expression" do
+    expect(grok).to pass
+  end
+
+  it "matches a simple message" do
+    expect(subject).to match(value)
+  end
+
+  it "generates the nagios_type field" do
+    expect(grok_match(subject, value)).to include("nagios_type" => "TIMEPERIOD TRANSITION")
+  end
+
+  it "generates the nagios_epoch field" do
+    expect(grok_match(subject, value)).to include("nagios_epoch" => "1427925600")
+  end
+
+  it "generates the nagios_esrvice field" do
+    expect(grok_match(subject, value)).to include("nagios_service" => "24X7")
+  end
+
+end
+
+describe "NAGIOSLOGLINE - SERVICE ALERT" do
+
+  let(:value)   { "[1427925689] SERVICE ALERT: varnish;Varnish Backend Connections;CRITICAL;SOFT;1;Current value: 154.0, warn threshold: 10.0, crit threshold: 20.0" }
+  let(:grok)    { grok_match(subject, value) }
+
+  it "a pattern pass the grok expression" do
+    expect(grok).to pass
+  end
+
+  it "matches a simple message" do
+    expect(subject).to match(value)
+  end
+
+  it "generates the nagios_type field" do
+    expect(grok_match(subject, value)).to include("nagios_type" => "SERVICE ALERT")
```

Comment:
In here you can use the value created in line 113, no need to call again the pattern matching.